### ISSUE TITLE
Unable to limit gamemode permissions.

### ISSUE THREAD

**Issue by mibby** [EssentialsX/Essentials#79]
You are unable to limit what gamemodes players can switch into with the specific gamemode permission nodes.

When you only have `essentials.gamemode.survival` and `essentials.gamemode.creative` permission nodes, you can't change gamemodes with the command `/gms` or `/gmc` because it outputs "You do not have access to that command."

It requires you to have `essentials.gamemode` permission node. However, if you have that permission node, you can change into the adventure and spectator gamemodes.

**Comment by justwheel** [comment]
I can confirm I am also having this issue on b184. Would be nice to see this fixed!

**Comment by drtshock** [comment]
I thought I fixed this but it looks like I wiped my changes. Will look into this soon.

**Comment by justwheel** [comment]
Thanks, looking forward for a fix!

**Comment by TomLewis** [comment]
Confirmed not working, im trying to give spectate permission to my mods.

**Comment by justwheel** [comment]
Likewise, that was the only one I've tried so far. I haven't actually tested the creative and survival permissions, but I feel like it would likely be the same for all of them. 

Cheers,
————————————

Justin W. Flory
jflory7@gmail.com
Sent via Android

----- Reply message -----

**Comment by drtshock** [comment]
Fixed. Read commit message / wiki for details :)

**Comment by mibby** [comment]
@drtshock This is not fixed. Please re-open the issue.
EssentialsX dev 189

With only the permission nodes `essentials.gamemode.survival` and `essentials.gamemode.creative`, I am unable to change my gamemode whatsoever. Adding the permission node `essentials.gamemode.all` does nothing. I have to re-add the permission `essentials.gamemode` to be able to change gamemodes, but this allows me to enter spectator and adventure gamemodes, even with `essentials.gamemode.all` not included.

Basically the commit did nothing.

**Comment by drtshock** [comment]
I see. I missed putting the check in the gamemode self and only put it in gamemode other players. Will commit this in like 60 seconds after I recompile.

**Comment by drtshock** [comment]
There we go.

**Comment by mibby** [comment]
Confirmed fixed now. 

Requires the command `essentials.gamemode` to use the /gamemode | /gm command, in addition to requiring individual gamemode permissions (such as `essentials.gamemode.survival`) or permission to access all.

**Comment by justwheel** [comment]
Confirmed fixed on my end too, thanks!
